lib/facter/odoo.rb

```diff
@@ -1,31 +1,70 @@
 require 'facter'
+require 'tempfile'
 
 Facter.add(:odoo) do
   setcode do
     if Facter::Util::Resolution.which('odoo')
-      major = minor = date = nil
-
-      begin
-        output = Facter::Core::Execution.execute('dpkg-query -W -f=\'${Version}\n\' odoo')
-        /^(?<major>\d+)\.(?<minor>\d+)(?:\+e)?\.(?<date>\d+)$/ =~ output
-      rescue Facter::Core::Execution::ExecutionFailure
-        output = nil
+      if Process.gid.zero?
         begin
-          output = Facter::Core::Execution.execute('pip3 show odoo')
-        rescue Facter::Core::Execution::ExecutionFailure
-          output = Facter::Core::Execution.execute('pip show odoo')
+          u = Etc.getpwnam('odoo') # an exception will be raised if no such user exist
+          Process.gid = Process.egid = u.gid
+          Process.uid = Process.euid = u.uid
```

Comment:
I guess this is causing trouble when execution continue.  We need to fork before changing identity, and wait for our child to complete.  Then the test suite might pass.

Reply:
Seems to help!